ios/RNGoogleSignin/RNGoogleSignin.m

```diff
@@ -1,20 +1,37 @@
 #import "RNGoogleSignin.h"
-#import "RCTEventDispatcher.h"
+// #import <React/RCTLog.h>
 
+@interface RNGoogleSignin ()
 
-@implementation RNGoogleSignin
+@property (nonatomic, strong) RCTPromiseResolveBlock currentUserAsyncResolve;
+@property (nonatomic, strong) RCTPromiseRejectBlock currentUserAsyncReject;
+@property (nonatomic, strong) RCTPromiseResolveBlock signInResolve;
+@property (nonatomic, strong) RCTPromiseRejectBlock signInReject;
+@property (nonatomic, strong) RCTPromiseResolveBlock revokeAccessResolve;
+@property (nonatomic, strong) RCTPromiseRejectBlock revokeAccessReject;
 
-RCT_EXPORT_MODULE();
+@property (nonatomic, assign) BOOL isSigningInSilently;
```

Comment:
I believe there are unnecessary properties defined: there are 4 `RCTPromise*Block`s:

+@property (nonatomic, strong) RCTPromiseResolveBlock currentUserAsyncResolve;
+@property (nonatomic, strong) RCTPromiseRejectBlock currentUserAsyncReject;
+@property (nonatomic, strong) RCTPromiseResolveBlock signInResolve;
+@property (nonatomic, strong) RCTPromiseRejectBlock signInReject;

While we need only 2. Let me illustrate:
+@property (nonatomic, assign) BOOL isSigningInSilently;

`isSigningInSilently` property controls what `RCTPromiseResolveBlock`s will be used for resolving/rejecting the promise passed from JS.

The possible flows of the program are (lets focus on resolving the promises):

1a . signIn() is called: `self.signInResolve` is assigned and `self.isSigningInSilently = NO;`
2a . didSignInForUser() is called and because `self.isSigningInSilently == NO`, `self.signInResolve` is used for resolving the promise to JS. `self.currentUserAsyncResolve` is *ignored*


1b . currentUserAsync() is called: `self.currentUserAsyncResolve` is assigned and `self.isSigningInSilently = YES;`
2b . didSignInForUser() is called and because `self.isSigningInSilently == YES`, `self.currentUserAsyncResolve` is used for resolving the promise to JS. `self.signInResolve` is *ignored*

It follows that we only need two `RCTPromise*Block` and I don't see a reason to have 4 instance properties if two are enough.

Reply:
Guess what? You’re right! I’ll make the changes. I’m glad we can simplify this quite a lot. Cannot believe I didn’t catch this myself since this feels so obvious. :)